test/core/client_config/uri_parser_test.c

```diff
@@ -56,6 +56,56 @@ static void test_fails(const char *uri_text) {
   GPR_ASSERT(NULL == grpc_uri_parse(uri_text, 0));
 }
 
+static void test_query_parts() {
+  {
+    const char *uri_text = "http://foo/path?a&b=B&c=&#frag";
```

Comment:
Can you add something like `"foo=bar=baz&foobar=="` to the query string  in addition to the three cases you had ?
(Since you wrote a TODO item talking about the case of having multiple '=' in the value field, it would be good to have a test case)..